packages/node/src/assignment/assignment-filter.ts

```diff
@@ -0,0 +1,22 @@
+import { Assignment, AssignmentFilter } from 'src/assignment/assignment';
+import { DAY_MILLIS } from 'src/assignment/assignment-service';
+import { Cache } from 'src/util/cache';
+
+export const DEFAULT_FILTER_CAPACITY = 65536;
+
+export class InMemoryAssignmentFilter implements AssignmentFilter {
+  private readonly cache: Cache<number>;
+
+  constructor(size: number) {
+    this.cache = new Cache<number>(size, DAY_MILLIS);
+  }
+
+  public shouldTrack(assignment: Assignment): boolean {
```

Comment:
Lets make this an `async` function, here's why: We may choose to re-implement this filter in a non-in-memory cache. I.e. something like redis, or another external persistent data store. In this case, we must make a network request to read/write from the store, which necessitates asynchronicity in javascript. 

My rule of thumb is, if you might want to persist the data in the future, make the function async. (see `FlagConfigCache`).